src/CommonITILValidation.php

```diff
@@ -1287,6 +1322,45 @@ public static function rawSearchOptionsToAdd()
             ]
         ];
 
+        $tab[] = [
+            'id'                 => '195',
+            'table'              => User::getTable(),
+            'field'              => 'name',
+            'linkfield'          => 'users_id_substitute',
+            'name'               => __('Approver substitute'),
+            'datatype'           => 'itemlink',
+            'right'              => (static::$itemtype == 'Ticket' ?
+                ['validate_request', 'validate_incident'] :
+                'validate'
+            ),
+            'forcegroupby'       => true,
+            'massiveaction'      => false,
+            'joinparams' => [
+                'beforejoin'         => [
+                    'table'          => ValidatorSubstitute::getTable(),
+                    'joinparams'         => [
+                        'jointype'           => 'child',
+                        'condition'          => [
+                            'REFTABLE.substitution_start_date' => ['<=', $_SESSION['glpi_currenttime']],
+                            'REFTABLE.substitution_end_date' => ['>=', $_SESSION['glpi_currenttime']],
+                        ],
+                        'beforejoin'         => [
+                            'table'              => User::getTable(),
+                            'linkfield'          => 'users_id_validate',
+                            'joinparams'             => [
+                                'beforejoin'             => [
+                                    'table'                  => static::getTable(),
+                                    'joinparams'                 => [
+                                        'jointype'                   => 'child',
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ],
+            ]
+        ];
```

Comment:
Search option to search an ITIL object by its validator, taking into accounts the substitutes of this validator, and the date range of effect of the delegation. Related to #12486